src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -2878,6 +2908,16 @@ var projectUniqueNamesForBuildIntToUpdate
             var result = new List<ResolvedAction>();
 
             var lockFileLookup = new Dictionary<string, LockFile>(PathUtility.GetStringComparerBasedOnOS());
+            PackageSourceMappingProvider packageSourceMappingProvider = null;
+            IReadOnlyList<PackageSourceMappingSourceItem> originalPackageSourceMappings = null;
+
+            if (newMappingID != null && newMappingSource != null)
+            {
+                packageSourceMappingProvider = new PackageSourceMappingProvider(Settings, shouldSkipSave: true);
```

Comment:
If I'm understanding this correctly, it seems the responsibility of `PackageSourceMappingProvider` is to mutate the settings to add or remove details about Package Source Mappings based on supplied parameters about package source mappings and decide whether to persist the settings or not. If that is the case, the name PackageSourceMappingProvider doesn't seem to fit its responsibility. It also appears that the details about package source mapping items is actually provided by an outside source for most or all of this class' methods. I'd like to understand the overall intention of this class better before I could suggest an alternate name. Happy to chat offline or in the next PR review meeting.

Reply:
Yes, our providers typically handle loading and manipulation of objects. For example, package sources have a provider here: https://github.com/NuGet/NuGet.Client/blob/63edda646848553467639abe6bbade5ebc5dec61/src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSourceProvider.cs#L464